sensors.py

```diff
@@ -13,35 +15,62 @@ class SensorReadingError(Exception):
     pass
 
 
+class MutableBool:
+    def __init__(self, initial_value: bool) -> None:
+        self.bool_value = initial_value
+
+    def __bool__(self) -> bool:
+        return self.bool_value
+
+    def __eq__(self, __value: object) -> bool:
+        return self.bool_value == __value
+
 class Sensor(ABC):
     def __init__(self) -> None:
         super().__init__()
         self._lock = Lock()
+        # MutableBool is true if values are new(weren't read since last addition)
+        bool_deque_type = tuple[MutableBool, deque[int | float]]
+        self._readings: DefaultDict[SensorType, bool_deque_type] = defaultdict(lambda: (MutableBool(False), deque(list(), 6)))
 
     @abstractmethod
     def get_reading(self, sensor_type: SensorType) -> int | float:
         raise NotImplementedError
 
+    def _add_reading(self, sensor_type: SensorType, value: int | float) -> None:
+        is_new, readings_deque = self._readings[sensor_type]
+        readings_deque.append(value)
+        is_new.bool_value = True
+
+    def _get_median(self, sensor_type: SensorType) -> int | float:
+        is_new, readings_deque = self._readings[sensor_type]
+        if not is_new or len(readings_deque) != readings_deque.maxlen:
+            raise SensorReadingError
+        is_new.bool_value = False
+        return median_low(readings_deque)
+
 
 class DHT(Sensor):
     def __init__(self) -> None:
         super().__init__()
         self.dht = DHT22(board.D4)
 
-    def get_reading(self, sensor_type: SensorType) -> int:
+    def get_reading(self, sensor_type: SensorType) -> int | float:
+        assert sensor_type in [SensorType.TEMPERATURE, SensorType.HUMIDITY], f"Wrong DHT sensor type({sensor_type})"
```

Comment:
I like these, but maybe we should have `self.sensor_types` in Sensor? and then pass in constructor?
Consider making it a separate commit.